brain/synapses.h

```diff
@@ -205,16 +205,25 @@ class Synapses
     BRAIN_API const float* decays() const;
 
     /** @return the absolute synaptic efficacies in millivolts. */
-    BRAIN_API const int* efficacys() const;
+    BRAIN_API const int* efficacies() const;
     //@}
 
+protected:
+    // The Impl pointer needs to be used in the Python wrapping as a custodian.
+    // Instead of moving the Impl declaration outside the implementation, this
+    // base class is used to give access to a virtual destructor, which is all
+    // the wrapping needs.
+    struct BaseImpl
+    {
+        virtual ~BaseImpl() {};
+    };
+    struct Impl;
```

Comment:
that forward declare is not need then.
Another option would be to declare a Impl pointer/reference, which is initialized in the ctor and all the casts can go away then. At the cost of storing an alias pointer which should not matter in terms of overhead.